File: src/main/java/com/github/alexandrecarlton/idea/settings/applier/impl/tools/ToolsSettingsApplier.java
package com.github.alexandrecarlton.idea.settings.applier.impl.tools;

import com.github.alexandrecarlton.idea.settings.applier.api.SettingsApplier;
import com.github.alexandrecarlton.idea.settings.layout.tools.ToolsSettings;
import com.github.alexandrecarlton.idea.settings.layout.tools.file_watchers.FileWatcherSettings;

import javax.inject.Inject;

public class ToolsSettingsApplier implements SettingsApplier<ToolsSettings> {

  private final SettingsApplier<FileWatcherSettings> fileWatcherSettingsApplier;

  @Inject
  public ToolsSettingsApplier(SettingsApplier<FileWatcherSettings> fileWatcherSettingsApplier) {
    this.fileWatcherSettingsApplier = fileWatcherSettingsApplier;
  }

  @Override
  public void apply(ToolsSettings settings) {
    settings.fileWatchers().forEach(fileWatcherSettingsApplier::apply);
  }

}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/applier/impl/tools/ToolsSettingsApplier.kt
package com.github.alexandrecarlton.idea.settings.applier.impl.tools

import com.github.alexandrecarlton.idea.settings.applier.api.SettingsApplier
import com.github.alexandrecarlton.idea.settings.layout.tools.ToolsSettings
import com.github.alexandrecarlton.idea.settings.layout.tools.file_watchers.FileWatcherSettings

import javax.inject.Inject

class ToolsSettingsApplier @Inject
constructor(private val fileWatcherSettingsApplier: SettingsApplier<FileWatcherSettings>) : SettingsApplier<ToolsSettings> {

    override fun apply(settings: ToolsSettings) {
        settings.fileWatchers?.forEach(fileWatcherSettingsApplier::apply)
    }
}
